### ISSUE TITLE
Phi for sequence classfication gives nan 

### ISSUE THREAD

**Issue by Alymostafa** [huggingface/transformers#28234]
### System Info

I tried fine-tuning the Phi-1_5 model for a sequence classification task, but it gives ```nan``` in the validation loss and ```no log``` in the training loss. I tried fp16 & float 32, and both have the same output. 
@younesbelkada 

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

```
model = AutoModelForSequenceClassification.from_pretrained(model_checkpoint,torch_dtype=torch.float32, device_map="cuda", num_labels=1, trust_remote_code=True)

args = TrainingArguments(
    f"{model_name}-finetuned-{task}",
    evaluation_strategy = "steps",
    eval_steps=100,
    save_strategy = "steps",
    learning_rate=2e-5,
    per_device_train_batch_size=4,
    per_device_eval_batch_size=4,
    num_train_epochs=6,
    fp16=True,
    load_best_model_at_end=True,
    metric_for_best_model=metric_name,
)


trainer = Trainer(
    model,
    args,
    train_dataset=encoded_dataset["train"],
    eval_dataset=encoded_dataset['validation'],
    tokenizer=tokenizer,
    compute_metrics=compute_metrics_for_regression
)

trainer.train()

### Expected behavior

loss as a numbers not nan or no logging

**Comment by younesbelkada** [comment]
Hi @Alymostafa 
I suspect there might be something off with the trust_remote_code version of phi_1-5, can you try with the official transformers version which is pushed here: https://huggingface.co/susnato/phi-1_5_dev ?
Can you also try to train with bf16 mixed precision? i.e. you need to load the model in fp32 and pass `bf16=True` in `TrainingArguments`

**Comment by Hambaobao** [comment]
> Hi @Alymostafa
> I suspect there might be something off with the trust_remote_code version of phi_1-5, can you try with the official transformers version which is pushed here: https://huggingface.co/susnato/phi-1_5_dev ?
> Can you also try to train with bf16 mixed precision? i.e. you need to load the model in fp32 and pass `bf16=True` in `TrainingArguments`

Hello, may I ask: is it necessary to load model in fp32 for amp?

**Comment by younesbelkada** [comment]
I think it would work for both scenoarios (loading it in fp32 or fp16) but is preferable to load it in fp32

**Comment by Hambaobao** [comment]
> I think it would work for both scenoarios (loading it in fp32 or fp16) but is preferable to load it in fp32

Thanks.👍

**Comment by Alymostafa** [comment]
@younesbelkada Thanks, it works.
 Is it possible to add a note on the Microsoft repo page to use the model you mentioned? That would be useful for other people.
